netsim/extra/repeater/plugin.py

```diff
@@ -0,0 +1,83 @@
+from box import Box
+from netsim import data
+from netsim.utils import log,strings
+from netsim.augment import links
+
+def clone_interfaces(link_data: Box, nodename: str, clones: list[str]) -> list[Box]:
+  """
+  update_ifindex - update any custom ifindex on the given interface
+  """
+  def update_ifindex(intf:Box, c:int) -> Box:
+    ifindex = intf.get('ifindex',None)
+    if ifindex is not None:
+      intf = data.get_box(intf)                                             # Need to make a fresh copy
+      intf.ifindex = ifindex + c + 1
+    return intf
+  
+  ifs = []
+  if nodename in [ i.node for i in link_data.interfaces ]:
+     link_data.pop('_linkname',None)
+     atts = [ i for i in link_data.interfaces if i.node==nodename ][0]      # Copy any interface atts
+     for c,clone in enumerate(clones):
+       l = data.get_box(link_data)
+       l.interfaces = [ atts+{ 'node': clone } ] + [ update_ifindex(i,c) for i in link_data.interfaces if i.node!=nodename ]
+       ifs.append(l)
+  return ifs
+
+def update_vlan_access_links(topology: Box, nodename: str, clones: list) -> None:
```

Comment:
This illustrates the major problem of this approach (not that I think it could be done in some other way): you're playing whack-a-mole with other data structures. You handled VLAN links (because you needed them), but there are also VRF links, LAG links, link groups, and topology components...

At the very minimum, the limitations have to be well-documented.

FWIW, not using the original node name (replacing "h" with "h_00", for example) would also uncover all cases where the cloning does not work as the node name used on a link would not be valid.

Reply:
Done, we now get errors when an unsupported construct is used in the topology (as it doesn't get updated)